locations/spiders/mattress_firm_us.py

```diff
@@ -0,0 +1,42 @@
+import json
+
+from locations.pipelines.phone_clean_up import PhoneCleanUpPipeline
+from locations.storefinders.where2getit import Where2GetItSpider
+
+
+class MattressFirmUSSpider(Where2GetItSpider):
+    name = "mattress_firm_us"
+    api_endpoint = "https://api.slippymap.com/mattressfirmsites/rest/getlist"
+    api_key = "88FD3C6E-2B22-11EE-86CD-EF1E9DC6E625"
+
+    def parse_item(self, item, location, **kwargs):
+        if location["mattress_firm_clearance_center"] == "yes":
+            item["brand"] = "Mattress Firm Clearance"
+            item["brand_wikidata"] = "Q6791878"
+            item["nsi_id"] = "mattressfirmclearance-032196"
+        elif location["sleep_expert_store"] == "yes":
+            item["brand"] = "Sleep Experts"
+            item["brand_wikidata"] = "Q7539688"
+            item["nsi_id"] = "sleepexperts-032196"
+        else:
+            item["brand"] = "Mattress Firm"
+            item["brand_wikidata"] = "Q6791878"
+            item["nsi_id"] = "mattressfirm-032196"
+        item["extras"]["shop"] = "bed"
```

Comment:
Can you use the `apply_category()` and `Categories` helpers for this? See https://github.com/alltheplaces/alltheplaces/blob/be9ae602e7dfe2240d2ef7d9addc918d70f86282/locations/spiders/circle_k_se.py#L21 for example.

Reply:
`shop=bed` isn't available in the `Categories` enum. Should I add it? Or maybe remove this line (and `item["nsi_id"] = …`) and let the NSI pipeline apply the shop tag

Reply:
Yep you can add it to Categories. Letting NSI add it makes sense too. I think in previous discussions we decided that setting nsi_id is unproductive (@Cj-Malone do you remember where that conversation happened?)

Reply:
Setting it disables the pipeline. But NSI ids aren't stable, that's why we went with matching in the first place, not hardcoding.

I can put it on my TODO list to write some docs about this.

Reply:
Ah sorry I forgot—the reason I added the category and NSI ID manually is that the matcher can't tell the difference between "Mattress Firm Clearance" and "Mattress Firm," because they have the same QID, and so matches neither.